### PR TITLE
Split DefaultConfigValidator to SPI implements

### PR DESCRIPTION
**Pull request by namelessssssssssss**
## What is the purpose of the change
**As a part of dubbo-core refactor, we need to reduce DefaultConfigValidator's dependency on other modules.**

Example:
Some config validation operation depends on certain module APIs
```Java
//ConfigValidatorUtils
 public static void validateAbstractInterfaceConfig(AbstractInterfaceConfig config) {
        checkName(LOCAL_KEY, config.getLocal());
        checkName("stub", config.getStub());
        checkMultiName("owner", config.getOwner());
        checkExtension(config.getScopeModel(), ProxyFactory.class, PROXY_KEY, config.getProxy());
        checkExtension(config.getScopeModel(), Cluster.class, CLUSTER_KEY, config.getCluster());
        checkMultiExtension(config.getScopeModel(), Arrays.asList(Filter.class, ClusterFilter.class), FILTER_KEY, config.getFilter());
        checkNameHasSymbol(LAYER_KEY, config.getLayer());

        List<MethodConfig> methods = config.getMethods();
        if (CollectionUtils.isNotEmpty(methods)) {
            methods.forEach(ConfigValidationUtils::validateMethodConfig);
        }
    }
...

```
After refactor:
```Java
@Activate
public class InterfaceConfigValidator implements ConfigValidator<AbstractInterfaceConfig> {

    public static void validateAbstractInterfaceConfig(AbstractInterfaceConfig config) {
        ConfigValidationUtils.checkName(LOCAL_KEY, config.getLocal());
        ConfigValidationUtils.checkName("stub", config.getStub());
        ConfigValidationUtils.checkMultiName("owner", config.getOwner());

        ConfigValidationUtils.checkExtension(config.getScopeModel(), ProxyFactory.class, PROXY_KEY, config.getProxy());
        ConfigValidationUtils.checkExtension(config.getScopeModel(), Cluster.class, CLUSTER_KEY, config.getCluster());
        ConfigValidationUtils.checkMultiExtension(config.getScopeModel(), Arrays.asList(Filter.class, ClusterFilter.class), FILTER_KEY, config.getFilter());
        ConfigValidationUtils.checkNameHasSymbol(LAYER_KEY, config.getLayer());

        List<MethodConfig> methods = config.getMethods();
        if (CollectionUtils.isNotEmpty(methods)) {
            methods.forEach(MethodConfig::validate);
        }
    }

    @Override
    public void validate(AbstractInterfaceConfig config) {
        validateAbstractInterfaceConfig(config);
    }

    @Override
    public boolean isSupport(Class<?> configClass) {
        return AbstractInterfaceConfig.class.isAssignableFrom(configClass);
    }
}

```
Then, we can put this ConfigValidator to any module we want.

## Brief changelog
* Turn interface ConfigValidator to SPI 
* Split DefaultConfigValidator to ConfigValidator implements.

This allows the configuration validation logic aggregated in DefaultConfigValidator to be distributed back into the ConfigValidator implements, reducing DefaultConfigValidator's dependency on other modules.

### CODE REVIEW

**Comment by chickenlj** [comment]
LGTM.

File: dubbo-common/src/main/java/org/apache/dubbo/common/constants/CommonConstants.java [hunk 1]
```diff
@@ -337,6 +337,8 @@ public interface CommonConstants {
 
     String REGISTRY_PROTOCOL_LISTENER_KEY = "registry.protocol.listener";
 
+    String REGISTRY_SERVICE_CLASS_NAME = "org.apache.dubbo.registry.RegistryService";
```
[the remark this answers is not in the document]
Comment (namelessssssssssss) [reply dubbo-common/src/main/java/org/apache/dubbo/common/constants/CommonConstants.java:340]: This constant used to replace direct reference of `RegistryService` in `ConfigValidationUtils`

Old:
```Java
//ConfigValidationUtils.loadRegistries:214
                    map.put(PATH_KEY, RegistryService.class.getName());
```

New:
```Java
//ConfigValidationUtils.loadRegistries:127
                    map.put(PATH_KEY,REGISTRY_SERVICE_CLASS_NAME);
```

File: dubbo-common/src/main/java/org/apache/dubbo/config/context/AbstractConfigManager.java [hunk 2]
```diff
@@ -613,7 +612,7 @@ protected <T extends AbstractConfig> boolean isNeedValidation(T config) {
 
     private ConfigValidator getConfigValidator() {
```
Comment (AlbumenJ) [inline dubbo-common/src/main/java/org/apache/dubbo/config/context/AbstractConfigManager.java:613]: Use `ConfigValidatorFacade` here

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

**Comment by namelessssssssssss** [comment]
Some IT need fix after this PR merge:
```Java
//MetadataIt.test4
        } catch (IllegalArgumentException e) {  //TODO ->  catch (ConfigValidationException e)
            Assert.assertTrue(e.getMessage().contains("Please specify valid protocol or address for metadata report"));
        }
```

**Review by AlbumenJ — APPROVED** [review]

**Review by AlbumenJ — APPROVED** [review]

**Pull request merged**